### ISSUE TITLE
React Native (monorepo) - Cannot Resolve ./index

### ISSUE THREAD

**Issue by airowe** [nrwl/nx#16666]
### Current Behavior

Runtime error building iOS app after upgrading to React Native 0.69.4 from 0.67.1, NX to 14.7.5, and React to 18.0.0.

### Expected Behavior

The app should be able to be run like it was prior to the upgrade

### Steps to Reproduce

1. Run the app, see the error

### Nx Report

```shell
Node : 16.13.2
   OS   : darwin arm64
   npm  : 8.19.2

   nx : 14.8.0
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.7.2
   @nrwl/detox : 14.7.5
   @nrwl/devkit : 15.7.2
   @nrwl/eslint-plugin-nx : 14.7.5
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.7.5
   @nrwl/js : 14.7.5
   @nrwl/linter : 14.7.5
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.7.5
   @nrwl/react-native : 14.7.5
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.7.5
   @nrwl/web : 14.7.5
   @nrwl/workspace : 14.7.5
   typescript : 4.8.4
```

### Failure Logs

```shell
Error: Cannot resolve ./index
    at .../node_modules/@nrwl/react-native/plugins/metro-resolver.js:30:15
    at Object.resolve (.../node_modules/react-native/node_modules/metro-resolver/src/resolve.js:31:12)
    at ModuleResolver.resolveDependency (.../node_modules/react-native/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:115:31)
    at DependencyGraph.resolveDependency (.../node_modules/react-native/node_modules/metro/src/node-haste/DependencyGraph.js:231:43)
    at .../node_modules/react-native/node_modules/metro/src/lib/transformHelpers.js:129:24
    at Server._resolveRelativePath (.../node_modules/react-native/node_modules/metro/src/Server.js:1137:12)
    at async Server.requestProcessor [as _processBundleRequest] (.../node_modules/react-native/node_modules/metro/src/Server.js:464:37)
    at async Server._processRequest (.../node_modules/react-native/node_modules/metro/src/Server.js:420:9)
```

**Comment by jaysoo** [comment]
Can you try running `nx migrate 15.9.3` and follow the instructions to install packages + run migrations?

e.g.

```
nx migrate 15.9.3 # this will generate a migrations.json file
npm install
nx migrate --run-migrations
```

I can see devkit and cypress plugins being misaligned in version compared to other plugins.

**Comment by jaysoo** [comment]
Looks like the issue's been closed. Did you resolve it?

**Comment by airowe** [comment]
Sorry, yes @jaysoo. After raising the issue I realized that the `jsBundleURLForBundleRoot` was specified as `index` because I copied the new declarations from the react-native upgrade helper that comes with the default index file reference, but my bundle root is actually main.tsx. After changing, it's working.

I've had a number of issues with the nx migrate process, that is throwing me in to dependency cycles. Hopefully will get through it soon.
